### ISSUE TITLE
Some cover art images are unrecognized

### ISSUE THREAD

**Issue by v-fox** [jonaski/strawberry#298]
**Describe the bug**

Played some sweet tuned from newest PG's action/anime masterpiece, [Astral Chain](https://en.wikipedia.org/wiki/Astral_Chain), only to discover that big fancy album cover art, that I made from game's cover art, is not shown in Strawberry, as it is in Clementine. It's like it's not there at all !

**To Reproduce**

Make high-res cover art using GIMP and export it as JPG with all fancy options turned on.

**Expected behavior**

Hyperdimensional [Stand](https://jojo.fandom.com/wiki/Stand)-cops !

**System Information:**
 - Operating system: openSUSE Tumbleweed
 - Strawberry Version: [0.6.6](https://build.opensuse.org/package/show/home:X0F:branches:multimedia/strawberry)

**Additional context**

`file` says `/media/storage-h2T/music_games/!PlatinumGames collection/Astral Chain/Cover.jpg: JPEG image data, JFIF standard 1.01, resolution (DPI), density 120x120, segment length 16, progressive, precision 8, 1024x1024, components 3`

**Comment by jonaski** [comment]
Is it embedded cover art, or a JPG in the album directory?

**Comment by v-fox** [comment]
> Is it embedded cover art, or a JPG in the album directory?

A file in the album directory.
I even have put `front,Front,cover,Cover,folder,Folder` in filenames for art to make sure that case does not matter but it did not help. I suspect that the reason for this is advanced JPG encoding in GIMP. Some other apps also known to fail on them.

**Comment by jonaski** [comment]
Do you have the libqt5-qtimageformats package installed?

**Comment by v-fox** [comment]
> Do you have the libqt5-qtimageformats package installed?

Sure.
It's just [THAT one jpg](https://drive.google.com/open?id=18thirWuth29_goAYR9Cvb8LrQI9NLCeI) that doesn't load so far. In fact, I think github can't show/downconvert such files also. I remember needing to re-encode an edited screenshot as png to put in here.

**Comment by jonaski** [comment]
Can you send me a file that doesnt work?

**Comment by jonaski** [comment]
Works fine here. I use opensuse tumbleweed too

**Comment by v-fox** [comment]
> Works fine here. I use opensuse tumbleweed too

In the actual player ? Then I'm stumped. I even rescanned collection and restarted player but it still ignores that album's art. Nothing relevant in terminal output either.

Maybe it doesn't like the aforementioned file-path or something ? In particular, '!' character.

**Comment by jonaski** [comment]
Yes, in the playing now popup.
We use Qt for images, the qt image plugins. So something is wrong with your Qt plugins

**Comment by v-fox** [comment]
> Yes, in the playing now popup.
> We use Qt for images, the qt image plugins. So something is wrong with your Qt plugins

That's a dubious statement. Considering that you and I, apparently, are using exactly the same package from the same distro.

I even tried building both Qt plugins and the player in my repos against all the rest of my packages: still no change, no errors, no image. I even tried searching something in `strace -ff -o <strace.log>` but the only cover-relevant stuff I've got was some links to JPGs in /tmp/strawberry-cover-*.jpg and complains about missing files in ~/.local/share/strawberry/strawberry/albumcovers/ Doesn't even mention attempts to open "Cover.jpg" or external cover jpg files that it actually manages to show. Not on `grep -i -E 'Cover\.|\.jpg' <strace.log>*`, anyway.

It doesn't even write anything about covers into verbose log, like it does for mood-files and lyrics. By the way, my /tmp is almost empty tmpfs with `rw,nosuid,size=163204k,huge=advise` options.

**Comment by jonaski** [comment]
I can't pinpoint exactly where the issue is, but my best guess is it that it has to do with your Qt installation.
From your debugging it sounds like it finds the cover, since it loads it to /tmp when playing, but doesn't show it.

To check that the cover url is correct:
`
sqlite3 ~/.local/share/strawberry/strawberry/strawberry.db
select art_manual, art_automatic from songs where album = 'album name;
`
But you should also be able to see the picked cover art in the edit tag dialog on the summary tab.
Try manually unset and setting the cover directly from the album directory using the cover manager.

**Comment by v-fox** [comment]
The command, indeed,returns the correct file and, I just noticed, its thumbnail is correctly shown in "Collection" panel BUT song info panel and tag editing show it as empty ! When [any song from the album](https://drive.google.com/open?id=1W45ZxMwUXFTXd7QZvTLnuGBc7SnBRBid) is playing, setting it manually shows it but then it resets itself to empty as soon as song stops :/

**Comment by v-fox** [comment]
@jonaski I insist that you reconsider about this: clearly the images can show up just fine. The real issue is that they unset to empty on track's start or change even after saving in tag-editing menu. I suspect that's not the only affected album but haven't checked yet if there are images in directories of other albums with missing covers.

Maybe it's some issue with handling database.
Also, in OBS SQLITE3_FTS5 non-fatally fails during build with both default sqlite and my package fork that explicitly enables FTS3-5 for compatibility with Clementine. It may be caused by some changes in latest sqlite release. Not sure if it has any effect on database handling.

**Comment by jonaski** [comment]
I can't reproduce the problem here. I even tested with the same audio file and the same JPG that you provided here.
I also have done a lot of testing previously on the entire cover manager and updating of covers since I fixed lot's of bugs related to it and added lot's of improvements including better cover providers.
Why don't you try to do debug it yourself? You got C++ projects yourself and seem qualified for it.
The only thing I can think of right now is that it could be a problem with the QUrl conversion somewhere if you can special characters in the files.

FTS5 fails inside OBS because it's a CMake run check and there are probably restrictions with run time checks in the OBS, it works fine in my buildbot dockers. sqlite on opensuse has FTS5 support enabled so that doesn't have anything to do with it.

**Comment by v-fox** [comment]
Obscure trigger doesn't mean that an issue isn't there.

Unfortunately, C++ code goes right over my head. If C makes some kind of sense, C++'s objective hell does not which I recently confirmed for myself while attempting to backport some half-baked experimental vsync-handling patches for [my kwin package](https://build.opensuse.org/package/show/home:X0F:HSF/kwin5). The only true project I've got is kiwi-based LiveCD-building toolkit which was forked from scripts that OBS's iso-building web-UI once used. Which I should update already because of deterioration in testing/recovery Live-distros. Anything else may be for small simple changes or github misclicks.

I wrote full path to the file here in case that filepath handling was known to be sketchy buy I have much weirder name with mixes of multiple language type (like latin + cyrillic + hieroglyphs + special symbols and all kinds of brackets) and they seem to work fine.

What about the code that loads the image ? When set manually in tag-editing menu during playback image shows up (NOT when "Save" is pressed but when file selection dialogue closes itself !), so whatever code-path, it uses at that time, does work. But after that playback it no longer does which mean it does it differently from when playback starts and it's supposed to search for the image by itself. And that issue is NOT present in Clementine.
Also, aren't external images take significantly more time to show up than embedded ones ? Tracks with them also say "cover art is not set" at all times but then load it anyway after few seconds.

I just found another issue with that album or in playlist handling in general: after remaking the playlist by loading exported Clementine's m3u file (described in https://github.com/jonaski/strawberry/issues/307#issuecomment-555714222) everything near and after '-' character in titles was lost, such that a title like "Jena - Catastrophe" turned into just "Jena". File's tags were correct but were not read automatically on loading or playback, resaving them in editing window to force change in the playlist did not have affect until doing "update changed collection folders" (even thought nothing was really changed). But playlist naming glitches is old known Amarok/Clementine issue. So, I would not be surprised if cover loading is similarly subtly screwed up.

I know that FTS5 is enabled but [player's build-system says it's unavailable](https://build.opensuse.org/package/live_build_log/multimedia:apps/strawberry/openSUSE_Tumbleweed/x86_64) which means it can skip some functionality or use inferior versions for some functions, at least that's what usually happening in such situations. With Clementine, unavailability of FTS3 was once a fatal failure, from what I remember.

**Comment by jonaski** [comment]
When you say that the album covers are saved to the albumcovers directory as hashes it sounds like the song is added to the playlist from outside the collection.
Audio files that are added from outside the collection aren't found by scanning so no directories are scanned to pick up album covers either. Embedded art will work but not files not covers in directories.
Check that you got the vinyl icon on the "source" in the playlist and not the folder icon which indicates how the files was added.

**Comment by v-fox** [comment]
Pretty weird not to search for covers and mood-files for added files, even if they are not in the collection, even if they are not recorded in the database. BUT I have finally managed to fix the issue for myself:
* I updated to 0.6.7.
* Exported the playlist as m3u.
* Imported it back and removed the original.
* The new copy did not have titles cut off after '-' symbol and missing art started loading.

Yet, just updating or rescanning database had no effect, it's re-exporting that did it !
I enabled "Source" field just for curiosity and result was weird: all song have no icon whatsoever except for tag-less opus-files from Youtube (directories of which I've added to the collection) which had green folder with acoustic driver. Which is probably because of my custom icon theme. It may be not recognizing 'inheritance' of icon theme properly or I just don't have that particular icon anywhere.
